card/mod/04_settings/lib/card/setting.rb

```diff
@@ -35,34 +40,43 @@ def to_type_id type
     end
 
     # usage:
-    # setting_opts group:              :permission | :event | ...
-    #              position:           <Fixnum> (starting at 1, default: add to end)
+    # setting_opts group:        :permission | :event | ...
+    #              position:     <Fixnum> (starting at 1, default: add to end)
     #              rule_type_editable: true | false (default: false)
     #              restricted_to_type: <cardtype> | [ <cardtype>, ...]
     def setting_opts opts
       group = opts[:group] || :other
       @@groups[group] ||= []
-      if opts[:position]
-        if @@groups[group][opts[:position]-1]
-          @@groups[group].insert(opts[:position]-1, self)
+      set_position group, opts[:position]
+
+      @codename = opts[:codename] ||
+                  name.match(/::(\w+)$/)[1].underscore.to_sym
+      self.rule_type_editable = opts[:rule_type_editable]
+      self.restricted_to_type =
+        if opts[:restricted_to_type]
+          type_ids = [opts[:restricted_to_type]].flatten.map do |cardtype|
+                       to_type_id(cardtype)
+                     end
```

Comment:
`end` at 59, 21 is not aligned with `type_ids = [opts[:restricted_to_type]].flatten.map do |cardtype|` at 57, 10.